src/Tomcat/TomcatController.ts

```diff
@@ -283,34 +292,28 @@ export class TomcatController {
             setTimeout(() => this.setStarted(serverInfo, true), 500);
             await javaProcess;
             this.setStarted(serverInfo, false);
-            this.disposeResource(statusBarCommand);
-            this.disposeResource(statusBar);
+            this.disposeResources([statusBarCommand, statusBar]);
             watcher.close();
             if (serverInfo.needRestart) {
                 serverInfo.needRestart = false;
                 await this.startTomcat(serverInfo, appName, output);
             }
         } catch (err) {
             this.setStarted(serverInfo, false);
-            this.disposeResource(statusBarCommand);
-            this.disposeResource(statusBar);
+            this.disposeResources([statusBarCommand, statusBar]);
             if (watcher) { watcher.close(); }
             throw new Error(err.toString());
         }
     }
 
-    private disposeResource(resource: vscode.Disposable | undefined): void {
-        if (resource) {
-            resource.dispose();
+    private disposeResources(resources: vscode.Disposable[]): void {
```

Comment:
since this method only has one argument, how about making it as an open-ended function arguments:

```javascript
private disposeResources(...resources: vscode.Disposable[])
```

then no need to wrap arguments in `[]`